src/KlaviyoBase.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+namespace Klaviyo;
+
+use GuzzleHttp\Client;
+use Psr\Http\Message\ResponseInterface;
+use Klaviyo\Exception\KlaviyoException;
+use Klaviyo\Model\ProfileModel;
+
+abstract class KlaviyoBase
+{
+    /**
+     * Host and versions
+     */
+    const HOST = 'https://a.klaviyo.com/api/';
+    const V1 = 'v1';
+    const V2 = 'v2';
+
+    /**
+     * Request methods
+     */
+    const HTTP_GET = 'GET';
+    const HTTP_POST = 'POST';
+    const HTTP_PUT = 'PUT';
+    const HTTP_DELETE = 'DELETE';
+
+    /**
+     * Error messages
+     */
+    const ERROR_INVALID_API_KEY = 'Invalid API Key.';
+    const ERROR_RESOURCE_DOES_NOT_EXIST = 'The requested resource does not exist.';
+    const ERROR_NON_200_STATUS = 'Request Failed with HTTP Status Code: %s';
+    
+    /**
+     * Request options
+     */
+    const API_KEY_HEADER = 'api-key';
+    const API_KEY_PARAM = 'api_key';
+    const DATA = 'data';
+    const HEADERS = 'headers';
+    const JSON = 'json';
+    const PROPERTIES = 'properties';
+    const QUERY = 'query';
+    const TOKEN = 'token';
+
+    /**
+     * Shared endpoints
+     */
+    const METRIC = 'metric';
+    const METRICS = 'metrics';
+    const RENDER = 'render';
+    const SEND = 'send';
+    const TIMELINE = 'timeline';
+
+    /**
+     * Klaviyo API arguments
+     */
+    const COUNT = 'count';
+    const EMAILS = 'emails';
+    const PAGE = 'page';
+    const PROFILES = 'profiles';
+    const SINCE = 'since';
+    const SORT = 'sort';
+
+    /**
+     * @var string
+     */
+    protected $private_key;
+
+    /**
+     * @var string
+     */
+    protected $public_key;
+
+    /**
+     * Constructor method for Base class.
+     *
+     * @param string $public_key Public key (account ID) for Klaviyo account
+     * @param string $private_key Private API key for Klaviyo account
+     * @param string $host Base URI for API requests. Can be overridden for testing.
+     */
+    // TODO: I don't think we can ever actually define a different host like this. Only calling this through __get in the Klaviyo class. Would like to change for local dev for example.
+    public function __construct( $public_key, $private_key )
+    {
+        $this->public_key = $public_key;
+        $this->private_key = $private_key;
+        $this->client = new Client(['base_uri' => self::HOST]);
```

Comment:
This shouldn't be necessary. In order to instantiate the client we require a private api key. we validate that there's a public key when the track service is instantiated. Although I'm not seeing the client anywhere, it should be present in the klaviyo.php file

Reply:
Might not be the best way to do this but I believe I originally set it up this way because php kwargs suck so while you can have two kwargs in the signature of the client you can't ever just use the second one. I chose the private key because it is required in all but the track service so it was easy to then add additional logic in the constructor of just that service class. My other thought was to create an abstract class for all private key services that creates some constructor logic so we can verify there's a private key for all of those services (it's not an abstract class actually but I'm blanking on what it's called, you can't inherit multiple classes but you can "implement" the thing that I'm forgetting the name of).

Just some context to how I was thinking about this originally. @siddwarkhedkar this is your show now so you do whatever you think is best!